components/Countdown.js

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import PropTypes from "prop-types";
+import timediff from "timediff";
+
+class Countdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.updateCountdown = this.updateCountdown.bind(this);
+    this.dateToCountDownTo = props.initialDate;
+    this.currentDate = Date.now();
+    this.Difference = this.getDifferenceInDates();
```

Comment:
What's the point for storing those values as an instance variable? Let's simplify a bit by moving getting all derived values in the `render` method:

```js
render() {
  const { days, hours, minutes, seconds } = timediff(this.state.currentDate, this.props.initialDate, "D:H:m:S");
  ...
}
```

and remove all instance variables. The only thing you'll need to store in the state is the `currentDate`:

```js

constructor(super) {
  super();
  this.state = {
    currentDate: Date.now()
  };
}
```